src/libraries/System.Console/src/System/IO/KeyParser.cs

```diff
@@ -0,0 +1,409 @@
+                'P' => ConsoleKey.F1,
+                'p' => ConsoleKey.Insert, // rxvt
+                'Q' => ConsoleKey.F2,
+                'R' => ConsoleKey.F3,
+                'S' => ConsoleKey.F4,
+                's' => ConsoleKey.PageDown, // rxvt
+                'T' => ConsoleKey.F5, // VT 100+
+                'U' => ConsoleKey.F6, // VT 100+
+                'u' => ConsoleKey.NoName, // it should be Begin, but we don't have such (press Num5 in rxvt to reproduce)
+                'V' => ConsoleKey.F7, // VT 100+
+                'W' => ConsoleKey.F8, // VT 100+
+                'w' when isRxvt => ConsoleKey.Home,
+                'w' when !isRxvt => ConsoleKey.End,
+                'X' => ConsoleKey.F9, // VT 100+
+                'Y' => ConsoleKey.F10, // VT 100+
+                'y' => ConsoleKey.PageUp, // rxvt
+                'Z' => ConsoleKey.F11, // VT 100+
+                '[' => ConsoleKey.F12, // VT 100+
+                _ => default
+            };
+
+        // maps "^[[{character}" for SCO terminals, based on https://vt100.net/docs/vt510-rm/chapter6.html
+        static (ConsoleKey key, ConsoleModifiers modifiers) MapSCO(char character)
+            => character switch
+            {
+                'A' => (ConsoleKey.UpArrow, 0),
+                'B' => (ConsoleKey.DownArrow, 0),
+                'C' => (ConsoleKey.RightArrow, 0),
+                'D' => (ConsoleKey.LeftArrow, 0),
+                'F' => (ConsoleKey.End, 0),
+                'G' => (ConsoleKey.PageDown, 0),
+                'H' => (ConsoleKey.Home, 0),
+                'I' => (ConsoleKey.PageUp, 0),
+                _ when char.IsBetween(character, 'M', 'X') => (ConsoleKey.F1 + character - 'M', 0),
+                _ when char.IsBetween(character, 'Y', 'Z') => (ConsoleKey.F1 + character - 'Y', ConsoleModifiers.Shift),
+                _ when char.IsBetween(character, 'a', 'j') => (ConsoleKey.F3 + character - 'a', ConsoleModifiers.Shift),
+                _ when char.IsBetween(character, 'k', 'v') => (ConsoleKey.F1 + character - 'k', ConsoleModifiers.Control),
+                _ when char.IsBetween(character, 'w', 'z') => (ConsoleKey.F1 + character - 'w', ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '@' => (ConsoleKey.F5, ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '[' => (ConsoleKey.F6, ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '<' or '\\' => (ConsoleKey.F7, ConsoleModifiers.Control | ConsoleModifiers.Shift), // the Spec says <, PuTTy uses \.
+                ']' => (ConsoleKey.F8, ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '^' => (ConsoleKey.F9, ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '_' => (ConsoleKey.F10, ConsoleModifiers.Control | ConsoleModifiers.Shift),
+                '`' => (ConsoleKey.F11, ConsoleModifiers.Control | ConsoleModifiers.Shift),
```

Comment:
```suggestion
                ''' => (ConsoleKey.F11, ConsoleModifiers.Control | ConsoleModifiers.Shift),
```
This one looks to me as `'` on https://vt100.net/docs/vt510-rm/chapter6.html.

Reply:
Same here, but PuTTy uses '`' for it and it's the only Terminal that supports SCO emulation that I could find and test

https://serverfault.com/questions/429901/how-to-change-the-terminal-to-sco-compliant-in-ubuntu